<launch>

  <arg name="enable_logging" default="false"/>
  <arg name="enable_ground_truth" default="true"/>
  <arg name="paused" value="false"/>
  <param name='coverage_planner_names' type='str' value='Axel'/>

  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(find rotors_gazebo)/worlds/basic.world"/>
    <!-- <arg name="debug" value="true"/> -->
    <arg name="paused" value="$(arg paused)"/>
    <!-- <arg name="gui" value="false"/> -->
  </include>

  <group ns="Axel">
    <include file="$(find rotors_gazebo)/launch/spawn_mav.launch">
      <arg name="mav_name" value="Axel" />
      <arg name="model" value="$(find rotors_description)/urdf/hummingbird_generic_odometry_sensor.gazebo" />
      <arg name="enable_logging" value="$(arg enable_logging)" />
      <arg name="enable_ground_truth" value="$(arg enable_ground_truth)" />
      <arg name="log_file" value="Axel"/>
      <arg name="x" value="-2.0"/>
      <arg name="y" value="2.0"/>
    </include>
    <node name="lee_position_controller_node" pkg="rotors_control" type="lee_position_controller_node" output="screen">
      <rosparam command="load" file="$(find rotors_gazebo)/resource/lee_controller_hummingbird.yaml" />
      <rosparam command="load" file="$(find rotors_gazebo)/resource/hummingbird.yaml" />
      <remap from="odometry" to="odometry_sensor1/odometry" />
    </node>
    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" />
    <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher" />
    <node name='coverage_planner' pkg='quad_control' type='coverage_planner.py' output='screen'/>
    <param name='name' type='str' value='Axel'/>
    <param name='others_names' type='str' value='Bo'/>
    <param name='token' type='bool' value='true'/>
    <node name='coverage_router' pkg='quad_control' type='coverage_router.py' output='screen'/>
    <!--node name="planner" pkg="quad_control" type="rotors_to_goal_planner_node.py" output="screen"/-->
    <!--node name="rotors_planner_node" pkg="quad_control_antonio" type="rotors_point_sequence_planner_node.py" output="screen"/-->
    <!--node name="waypoint_publisher" pkg="rotors_gazebo" type="waypoint_publisher" output="screen" args="-1 -1 1 0 5"/-->
  </group>
  
  
  <group ns="Bo">
    <include file="$(find rotors_gazebo)/launch/spawn_mav.launch">
      <arg name="mav_name" value="Bo" />
      <arg name="model" value="$(find rotors_description)/urdf/hummingbird_generic_odometry_sensor.gazebo" />
      <arg name="enable_logging" value="$(arg enable_logging)" />
      <arg name="enable_ground_truth" value="$(arg enable_ground_truth)" />
      <arg name="log_file" value="Bo"/>
      <arg name="x" value="-2.0"/>
      <arg name="y" value="-2.0"/>
    </include>
    <node name="lee_position_controller_node" pkg="rotors_control" type="lee_position_controller_node" output="screen">
      <rosparam command="load" file="$(find rotors_gazebo)/resource/lee_controller_hummingbird.yaml" />
      <rosparam command="load" file="$(find rotors_gazebo)/resource/hummingbird.yaml" />
      <remap from="odometry" to="odometry_sensor1/odometry" />
    </node>
    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" />
    <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher" />
    <node name='coverage_planner' pkg='quad_control' type='coverage_planner.py' output='screen'/>
    <param name='name' type='str' value='Bo'/>
    <param name='others_names' type='str' value='Axel'/>
    <param name='token' type='bool' value='false'/>
    <node name='coverage_router' pkg='quad_control' type='coverage_router.py' output='screen'/>
    <!--node name="planner" pkg="quad_control" type="rotors_to_goal_planner_node.py" output="screen"/-->
    <!--node name="rotors_planner_node" pkg="quad_control_antonio" type="rotors_point_sequence_planner_node.py" output="screen"/-->
    <!--node name="waypoint_publisher" pkg="rotors_gazebo" type="waypoint_publisher" output="screen" args="-1 -1 1 0 5"/-->
  </group>
  
  <node name='coverage_plotter' pkg='quad_control' type='coverage_plotter.py' output='screen'/>
  <!--node name='rqt_plot' pkg='rqt_plot' type='rqt_plot'/-->
  
</launch>
